consensus/hotstuff/verification/combined_signer_v2.go

```diff
@@ -0,0 +1,123 @@
+package verification
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/signature"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+)
+
+// CombinedSignerV2 creates votes for the main consensus. Per protocol specification
+// a consensus participant can vote for a block by _either_ signing the block with
+// its staking key _or_ with its random beacon key.
+// If the participant provides signs with its random beacon key, it contributes to
+// hotstuff _and_ the random beacon. A staking signature _only_ contributes to
+// hotstuff. The latter is a useful fallback for a node that failed to properly
+// participate in the DKG.
+// The default behaviour of a node is to contribute to both the random beacon and
+// hotstuff, as it will yield higher rewards. Therefore, we always sign with the
+// random beacon key if it is available.
+// TODO: to be replaced by CombinedSignerV3 for mature V2 solution.
+// The difference between V2 and V3 is that V2 will sign 2 sigs, whereas
+// V3 only sign 1 sig.
+type CombinedSignerV2 struct {
+	staking           module.MsgSigner
```

Comment:
`MsgSigner` is also a thin layer around `Local` to re use its `Sign` method. I would just use `Local.Sign` straight away and avoid the extra module, but I have no strong opinion for this.